python/hail/expr/expression.py

```diff
@@ -3118,59 +3173,108 @@ def to_boolean(self):
 class CallExpression(Expression):
     """Expression of type :class:`.TCall`.
 
-    >>> call = functions.capture(Call(1))
+    >>> call = functions.capture(Call([0, 1]))
     """
-    @property
-    def gt(self):
-        """Returns the triangle number of :meth:`.Call.gtj` and :meth:`.Call.gtk`.
+
+    def __getitem__(self, item):
+        """Get the i*th* allele.
 
         Examples
         --------
+
+        Index with a single integer:
+
         .. doctest::
 
-            >>> eval_expr(call.gt)
+            >>> eval_expr(call[1])
             1
 
+            >>> eval_expr(call[0])
+            0
+
+        Parameters
+        ----------
+        item : int or :class:`.Int32Expression`
+            Allele index.
+
         Returns
         -------
         :class:`.Int32Expression`
-            Triangle number of the two alleles.
         """
-        return self._field("gt", TInt32())
+        if isinstance(item, slice):
+            raise NotImplementedError("CallExpression does not support indexing into array with a slice.")
+        else:
+            item = to_expr(item)
+            if not isinstance(item._type, TInt32):
+                raise TypeError("Call expects allele index to be an expression of type 'Int32', "
+                                "found expression of type '{}'".format(item._type))
+            return self._index(TInt32(), item)
 
-    def gtj(self):
-        """Returns the allele index of the first allele.
+    @property
+    def ploidy(self):
+        """Return the number of alleles of this call.
 
         Examples
         --------
         .. doctest::
 
-            >>> eval_expr(call.gtj())
-            0
+            >>> eval_expr(call.ploidy)
+            2
 
         Returns
         -------
         :class:`.Int32Expression`
-            First allele index.
         """
-        return self._method("gtj", TInt32())
+        return self._method("ploidy", TInt32())
 
-    def gtk(self):
-        """Returns the allele index of the second allele.
+    @property
+    def phased(self):
+        """True if the call is phased.
 
         Examples
         --------
         .. doctest::
 
-            >>> eval_expr(call.gtk())
-            1
+            >>> eval_expr(call.phased)
+            False
 
         Returns
         -------
-        :class:`.Int32Expression`
-            Second allele index.
+        :class:`.BooleanExpression`
+        """
+        return self._method("isPhased", TBoolean())
+
+    def is_haploid(self):
+        """True if the call has ploidy equal to 1.
+
+        Examples
+        --------
+        .. doctest::
+
+            >>> eval_expr(call.is_haploid())
+            False
+
+        Returns
+        -------
+        :class:`.BooleanExpression`
         """
-        return self._method("gtk", TInt32())
+        return self._method("isHaploid", TBoolean())
```

Comment:
You don't need the full range of expressions in the function registry if you can implement the python methods in terms of what's available.  For example, this could just be `return self.ploidy == 1`.